### PR TITLE
feat(colorloupe)!: token migration

### PR DESCRIPTION
**Pull request by jenndiaz**
## Description
BREAKING CHANGE: This migrates the ColorLoupe component to core tokens.
- includes adding template and story for ColorLoupe
- ColorLoupe is also displayed with ColorSlider and ColorHandle

## How and where has this been tested?
comparing local build of with figma for ColorLoupe and with https://spectrum.adobe.com/page/color-loupe/

**Browser(s) and OS(s) this was tested with:**
Chrome 109 on macOS
Safari 16.2 on macOS
Firefox 109 on macOS

- [x] Checked WHCM in assistive labs 

## To-do list

- [x] If my change impacts other components, I have tested to make sure they don't break.
- [x] If my change impacts documentation, I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).
- [x] I have tested these changes in Windows High Contrast mode.
- [x] received approval of assigned designer 

- [ ] This pull request is ready to merge.

### CODE REVIEW

File: components/tokens/custom-express/custom-large-vars.css [hunk 1]
```diff
@@ -15,4 +15,7 @@ governing permissions and limitations under the License.
 .spectrum--express.spectrum--large {
   --spectrum-colorwheel-path: "M 118 118 m -118 0 a 118 118 0 1 0 236 0 a 118 118 0 1 0 -236 0.2 M 118 118 m -92 0 a 92 92 0 1 0 184 0 a 92 92 0 1 0 -184 0";
   --spectrum-colorwheel-path-borders: "M 120 120 m -120 0 a 120 120 0 1 0 240 0 a 120 120 0 1 0 -240 0.2 M 120 120 m -90 0 a 90 90 0 1 0 180 0 a 90 90 0 1 0 -180 0";
+  --spectrum-colorloupe-express-loupe: url(#express-loupe-clip-l);
+  --spectrum-colorloupe-express-m-border: none;
+  --spectrum-colorloupe-express-l-border: block;
```
Comment (jenndiaz) [inline components/tokens/custom-express/custom-large-vars.css:20]: can I have the custom tokens in this PR or do I need to put them in separate PR like we did for ColorWheel.

File: components/colorarea/metadata/colorarea.yml [hunk 2]
```diff
@@ -24,12 +24,6 @@ examples:
 
         <div class="spectrum-ColorHandle spectrum-ColorArea-handle">
           <div class="spectrum-ColorHandle-color" style="background-color: rgb(255, 0, 0)"></div>
-          <svg class="spectrum-ColorLoupe">
```
Comment (jenndiaz) [inline components/colorarea/metadata/colorarea.yml:27]: We had this code here in ColorArea and[ ColorWheel ](components/colorwheel/metadata/colorwheel.yml)which both use ColorHandle and in turn can use ColorLoupe. However they did not have the `is-open` class so the ColorLoupe was not displayed and when toggled to be visible it does not display correctly in its current state (does not include path for express and does not handle selected color). 

I opted to remove it instead of adding it since there is demo of ColorLoupe within ColorHandle and ColorSlider and adding the new code here would increase the lines for each of these .yml files significantly (~550 lines).

**Review by jawinn — COMMENTED** [review]
Nice work! The only thing I am seeing is a difference in the outer border. Is this an expected difference?
color-loupe-compare
[new / old]

**Comment by jenndiaz** [comment]
> Nice work! The only thing I am seeing is a difference in the outer border. Is this an expected difference? color-loupe-compare [new / old]

Yes this is an expect change, new designs brought the white border to 2pixels and added an "inner border" of 1px

File: components/colorloupe/index.css [hunk 3]
```diff
@@ -1,53 +1,97 @@
...
   position: absolute;
-  transform: translate(0, var(--spectrum-colorloupe-animation-distance));
+  transform: translate(0, var(--mod-colorloupe-animation-distance, var(--spectrum-colorloupe-animation-distance)));
   opacity: 0;
   transform-origin: bottom center;
-  bottom: calc(50% + var(--spectrum-colorloupe-offset));
-  left: calc(50% - calc(var(--spectrum-colorloupe-width-adjusted) / 2));
+  inset-block-end: calc(var(--spectrum-colorhandle-size) + var(--mod-colorloupe-offset, var(--spectrum-colorloupe-offset)));
+  inset-inline: calc(40% - calc(var(--mod-colorloupe-width, var(--spectrum-colorloupe-width)) / 2));
   transition: transform 100ms ease-in-out, opacity 125ms ease-in-out;
```
Comment (castastrophe) [inline components/colorloupe/index.css:39]: Curious - should these be leveraging tokens?

```suggestion
  transition: transform 100ms ease-in-out, opacity 125ms ease-in-out;
```

File: components/colorloupe/index.css [hunk 4]
```diff
@@ -1,53 +1,97 @@
...
-  --spectrum-colorloupe-animation-distance: var(
-    --spectrum-global-dimension-static-size-100
-  );
+  --spectrum-colorloupe-drop-shadow-x: var(--spectrum-drop-shadow-x);
+  --spectrum-colorloupe-drop-shadow-y: var(--spectrum-color-loupe-drop-shadow-y);
+  --spectrum-colorloupe-drop-shadow-blur: var(--spectrum-color-loupe-drop-shadow-blur);
+  --spectrum-colorloupe-drop-shadow-color: var(--spectrum-color-loupe-drop-shadow-color);
+
+  --spectrum-colorloupe-outer-border-width: var(--spectrum-color-loupe-outer-border-width);
+  --spectrum-colorloupe-inner-border-width: 1px; /* TODO: replace with updated token for color-loupe-inner-border-width */
```
Comment (castastrophe) [inline components/colorloupe/index.css:24]: Is this in a yet-to-be-released tokens value?

File: components/colorloupe/index.css [hunk 5]
```diff
@@ -1,53 +1,97 @@
...
+.spectrum-colorloupe-express-m-borders {
+  display: var(--spectrum-colorloupe-express-m-border);
+}
+
+.spectrum-colorloupe-express-l-borders {
+  display: var(--spectrum-colorloupe-express-l-border);
+}
+
+/* hack: make sure we don't complain about this color being undefined */
+.🤫 {
```
Comment (castastrophe) [inline components/colorloupe/index.css:89]: haha gets me every time. I kind of love that it's an emoji.

File: components/colorloupe/index.css [hunk 6]
```diff
@@ -1,53 +1,97 @@
...
+
+.spectrum-colorloupe-express-m-borders {
+  display: var(--spectrum-colorloupe-express-m-border);
+}
+
+.spectrum-colorloupe-express-l-borders {
+  display: var(--spectrum-colorloupe-express-l-border);
+}
+
+/* hack: make sure we don't complain about this color being undefined */
```
Comment (castastrophe) [inline components/colorloupe/index.css:88]:
```suggestion
/* hack: make sure we don't complain about this color being undefined */
```

Maybe include a note here about it being a user-defined value that is dynamic and thus used by but not defined by the styles?

File: components/colorloupe/index.css [hunk 7]
```diff
@@ -1,53 +1,97 @@
...
+  fill: none;
+  stroke: var(--highcontrast-colorloupe-outer-border-color, var(--mod-colorloupe-outer-border-color, var(--spectrum-colorloupe-outer-border-color)));
+  stroke-width: calc(var(--mod-colorloupe-outer-border-width, var(--spectrum-colorloupe-outer-border-width)) + var(--mod-colorloupe-inner-border-width ,var(--spectrum-colorloupe-inner-border-width)));
 }
 
 .spectrum-ColorLoupe-spectrum {
   display: var(--spectrum-colorloupe-spectrum-visibility);
 }
+
+.spectrum-ColorLoupe-express {
```
Comment (castastrophe) [inline components/colorloupe/index.css:71]:
```suggestion
.spectrum-ColorLoupe-express {
```

Oh hmm - we've been trying to avoid express-specific naming in our components b/c ideally it will be driven entirely by component tokens being altered.  This is a tricky one.  I'd like to see this moved to the express file in themes somehow. Want to brainstorm on it today?

File: components/colorloupe/metadata/colorloupe.yml [hunk 8]
```diff
@@ -1,19 +1,34 @@
...
   - Apply `.is-open` to display the loupe
   - Color Loupe does not have a disabled style; do not show it when the handle its attached to is disabled.
 examples:
-  - id: color-area
+  - id: color-loupe
     name: Standard
     demoClassName: spectrum-CSSExample-example--spacious
     markup: |
-      <svg class="spectrum-ColorLoupe is-open" style="--spectrum-picked-color: rgba(255, 0, 0, 0.5); position: absolute; top: 25%; left: 50%;">
+      <svg class="spectrum-ColorLoupe is-open" overflow="visible" style="--spectrum-picked-color: rgba(255, 0, 0, 0.5); position: absolute; inset-block-start: 25%; inset-inline-start: 50%;">
```
Comment (castastrophe) [inline components/colorloupe/metadata/colorloupe.yml:13]: `overflow="visible"` <- legit did not know that worked on SVGs! #TIL

File: components/colorloupe/stories/template.js [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+    'inset-block-start': '25%',
+    'inset-inline-start': '50%',
+    overflow: "visible"
+  },
+  customClasses = [],
+  ...globals
+}) => {
+  const { express } = globals;
+
+  return html`
```
Comment (castastrophe) [inline components/colorloupe/stories/template.js:22]:
```suggestion
  return svg`
```

Since this whole thing is an SVG, you can switch from using the html tag literal to the svg tag literal if you like. I'm not sure it makes much of a difference but it might have efficiency gains

File: components/tokens/custom-express/custom-large-vars.css (hunk 1 above)
Comment (castastrophe) [reply components/tokens/custom-express/custom-large-vars.css:20]: Oh, I've no idea. If Patrick had it done in a separate PR then I guess, let's do that again just to be safe?

File: components/colorloupe/index.css (hunk 3 above)
Comment (jenndiaz) [reply components/colorloupe/index.css:39]: I had the same thought about `--spectrum-colorloupe-animation-distance: 8px;` and asked in spectrum_css_verifiation about it. 

https://adobedesign.slack.com/archives/GQ09RCBA5/p1678396366690899?thread_ts=1678388705.495289&cid=GQ09RCBA5 

Sounds like Garth has a project in the works to add tokens for animations.

File: components/colorloupe/metadata/colorloupe.yml (hunk 8 above)
Comment (jenndiaz) [reply components/colorloupe/metadata/colorloupe.yml:13]: These components have been a lot of SVG learnings 😆

File: components/colorloupe/stories/template.js (hunk 9 above)
Comment (jenndiaz) [reply components/colorloupe/stories/template.js]: got this updated!

File: components/colorloupe/index.css (hunk 6 above)
Comment (jenndiaz) [reply components/colorloupe/index.css]: yeah I like this, got it updated!

File: components/colorloupe/index.css (hunk 3 above)
Comment (castastrophe) [reply components/colorloupe/index.css:39]: Ah gotcha. Cool cool cool.

File: components/colorloupe/index.css (hunk 7 above)
Comment (castastrophe) [reply components/colorloupe/index.css:71]: @pfulton Adding you here for your thoughts as well!

Comment (pfulton) [reply components/colorloupe/index.css:71]: Yeah, I'm in agreement with @castastrophe - we can/should chat about this, but I'm thinking that SWC would also prefer that we handle situations like this via custom properties.

File: components/colorloupe/themes/spectrum.css [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+@container (--system: spectrum) {
+  .spectrum-ColorLoupe {
+    --spectrum-colorloupe-express-visibility: none;
+    --spectrum-colorloupe-spectrum-visibility: block;
```
Comment (pfulton) [inline components/colorloupe/themes/spectrum.css:15]: We'll chat about this, but it's related to my comment about handling things between Spectrum & Express with custom properties rather than classes that refer to those themes.

A somewhat related example is what I did with the Menu PR based upon feedback from the SWC team. I set up [three custom properties](https://github.com/adobe/spectrum-css/pull/1508/files#diff-3b8d6209262a97e225fba580d5a675267ced29e23aadabdcc2cb73332bf11d94R165-R167), setting the ["default" one](https://github.com/adobe/spectrum-css/pull/1508/files#diff-3b8d6209262a97e225fba580d5a675267ced29e23aadabdcc2cb73332bf11d94R167) to the "shown" state, and then later, I switch between the two by resetting their definitions [here](https://github.com/adobe/spectrum-css/pull/1508/files#diff-3b8d6209262a97e225fba580d5a675267ced29e23aadabdcc2cb73332bf11d94R300) and [here](https://github.com/adobe/spectrum-css/pull/1508/files#diff-3b8d6209262a97e225fba580d5a675267ced29e23aadabdcc2cb73332bf11d94R306).

**Review by pfulton — CHANGES_REQUESTED** [review]
Just a couple of notes here, @jenndiaz - let's chat about these soon! I'll reach out on Slack.

File: components/colorloupe/metadata/colorloupe.yml [hunk 11]
```diff
@@ -1,187 +1,43 @@
...
+            <use xlink:href="#path" fill="black" />
+          </mask>
+          <pattern id="checkerboard-primary" patternUnits="userSpaceOnUse" width="16" height="16" class="checkerboard-pattern">
+            <rect x="0" y="0" width="8" height="8" />
+            <rect x="8" y="8" width="8" height="8" />
+          </pattern>
+          <pattern id="checkerboard-secondary" patternUnits="userSpaceOnUse" width="20" height="20" class="checkerboard-pattern">
+            <rect x="0" y="0" width="10" height="10" />
+            <rect x="10" y="10" width="10" height="10" />
+          </pattern>
```
[the remark this answers is not in the document]
Comment (jenndiaz) [reply components/colorloupe/metadata/colorloupe.yml:27]: @pfulton I could use your thoughts on this

File: components/colorloupe/index.css [hunk 12]
```diff
@@ -1,53 +1,106 @@
...
+  --spectrum-colorloupe-drop-shadow-y: var(--spectrum-color-loupe-drop-shadow-y);
+  --spectrum-colorloupe-drop-shadow-blur: var(--spectrum-color-loupe-drop-shadow-blur);
+  --spectrum-colorloupe-drop-shadow-color: var(--spectrum-color-loupe-drop-shadow-color);
+
+  --spectrum-colorloupe-outer-border-width: var(--spectrum-color-loupe-outer-border-width);
+  --spectrum-colorloupe-inner-border-width: 1px; /* TODO: replace with updated token for color-loupe-inner-border-width */
+  --spectrum-colorloupe-outer-border-color: var(--spectrum-color-loupe-outer-border);
+  --spectrum-colorloupe-inner-border-color: var(--spectrum-color-loupe-inner-border);
+
+  /* --spectrum-colorloupe-checkerboard-size: var(--spectrum-opacity-checkerboard-square-size); */
```
Comment (castastrophe) [inline components/colorloupe/index.css:28]: Is this still needed?

**Review by pfulton — APPROVED** [review]

**Review by pfulton — APPROVED** [review]
